apis/v1/gateway_types.go

```diff
@@ -1230,6 +1247,59 @@ const (
 	ListenerReasonPending ListenerConditionReason = "Pending"
 )
 
+const (
+	// This condition indicates that TLS configuration within this Listener
+	// conflicts with TLS configuration in another Listener on the same port.
+	// This could happen for two reasons:
+	//
+	// 1) Overlapping Hostnames: Listener A matches *.example.com while Listener
+	//    B matches foo.example.com.
+	// B) Overlapping Certificates: Listener A contains a certificate with a
```

Comment:
is this a superset of overlapping hostname?

Reply:
if not, which should be reported if both are true

Reply:
Updated, thanks!